web/api/backend.go

```diff
@@ -113,45 +114,58 @@ func Handler(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		// If the structure of the body is wrong, return an HTTP error.
 		w.WriteHeader(http.StatusBadRequest)
+		io.WriteString(w, "Unable to Parse JSON")
 		return
 	}
 
-	// Checking if there is a token to use if the user asks to use a token or a credential body. Will return an error if those components are missing.
-	if (requestBody.UseToken && len(requestBody.Token) == 0) || (len(requestBody.Credential) == 0) {
-		w.WriteHeader(http.StatusBadRequest)
+	// JSON BODY representing the credentials attribute to be used (i.e. either the credentials found in the token or inputted).
+	creds := make(map[string]interface{})
+
+	// JSON Body representing the credentials attribute with the following format: "credentials":{"scopes":[],...}
+	// This is so to match with how the wrapper class will save the credentials file to memory.
+	credsString := make(map[string]interface{})
+	credsString = nil
+
+	// If command type is test or token, credentials are not necessary
+	if !(reflect.DeepEqual(requestBody.CommandType, "test")) && !(reflect.DeepEqual(requestBody.CommandType, "info")) && !(reflect.DeepEqual(requestBody.CommandType, "reset")) {
+		// Checking if there is a token to use if the user asks to use a token or a credential body. Will return an error if those components are missing.
+		if (requestBody.UseToken && len(requestBody.Token) == 0) || (!requestBody.UseToken && len(requestBody.Credential) == 0) {
+			w.WriteHeader(http.StatusBadRequest)
+			if requestBody.UseToken {
+				io.WriteString(w, "missing token file")
```

Comment:
We should capitalize http response errors here and elsewhere.

Reply:
dones